plugins/modules/v_switch.py

```diff
@@ -0,0 +1,487 @@
+        and str(server['server_number']) not in desired_servers
+    ]
+
+
+def get_servers_to_add(current_servers, desired_servers):
+    current_ids = [str(server['server_number']) for server in current_servers]
+    current_ips = [server['server_ip'] for server in current_servers]
+    return [
+        server
+        for server in desired_servers
+        if server not in current_ips and server not in current_ids
+    ]
+
+
+def set_desired_servers(module, id_):
+    v_switch = get_v_switch(module, id_)
+    changed = False
+
+    servers_to_delete = get_servers_to_delete(v_switch['server'], module.params['servers'])
+    if len(servers_to_delete) > 0:
+        v_switch = delete_servers(module, id_, servers_to_delete)
+        changed = True
+    servers_to_add = get_servers_to_add(v_switch['server'], module.params['servers'])
+    if len(servers_to_add) > 0:
+        v_switch = add_servers(module, id_, servers_to_add)
+        changed = True
+    return (v_switch, changed)
+
+
+def main():
+    argument_spec = dict(
+        vlan=dict(type='int', required=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        servers=dict(type='list', elements='str', default=[]),
+        wait=dict(type='bool', default=True),
+        wait_delay=dict(type='int', default=10),
+        timeout=dict(type='int', default=180),
+    )
+    argument_spec.update(ROBOT_DEFAULT_ARGUMENT_SPEC)
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    v_switches, error = fetch_url_json(module, V_SWITCH_BASE_URL, accept_errors=['UNAUTHORIZED'])
+
+    if error:
+        module.fail_json(msg='Please check your current user and password configuration')
+
+    matched_v_switches = [
+        v
+        for v in v_switches
+        if v['name'] == module.params['name'] and v['vlan'] == module.params['vlan']
+    ]
+    non_cancelled_v_switches = [m for m in matched_v_switches if m['cancelled'] is False]
+    result = {'changed': False}
+
+    if len(non_cancelled_v_switches) > 1:
+        module.fail_json(
+            msg='Multiple vSwitches with same name and VLAN ID in non cancelled status. Clean it'
+        )
+
+    elif len(non_cancelled_v_switches) == 1:
+        id_ = non_cancelled_v_switches[0]['id']
+        v_switch, changed = set_desired_servers(module, id_)
+        if changed:
+            result['changed'] = True
+
+        if module.params['state'] == 'present':
+            result['v_switch'] = v_switch
+        elif module.params['state'] == 'absent':
+            delete_v_switch(module, id_)
+            result['changed'] = True
+        else:
+            # not reachable
+            raise NotImplementedError
+    else:
+        if module.params['state'] == 'present':
+            v_switch = create_v_switch(module)
+            result['changed'] = True
+            if module.params['servers']:
```

Comment:
This `if` only makes sense when you are testing for `None`, and there is not `[]` as a default.
```suggestion
            if module.params['servers'] is not None:
```

Reply:
I don't get your point. If you have any servers on the list you will have to add them, if there are nothing or no servers there then you should skip.
I mean, empty list or None should be skipped.

Reply:
Empty list and `None` are two different values that should be treated differently.

If I set the parameter explicitly to an empty list I would expect all servers to be removed.

If I do not set the parameter (value `None`) then I would expect the module to not touch the server list.

Reply:
I agree with you in that and it was implemented in that way.

In line `477` this logic is just for the case that it is not exists and should exists so create it. In that case is the same if there are `None` or `[]`  because we are not going to add any servers in both options.

If you have another point, please insists because I cannot see what is wrong there.